commit 06c5a5314bfe34efdef107cc9a928987c025f11f (refs/changes/66/1466/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-02-08T00:06:59+00:00 (2 years, 8 months ago)
    
    joyent/node-cueball#88 Uncaught Error: Invalid FSM transition: backoff => closed
    Reviewed by: Trent Mick <trent.mick@joyent.com>

diff --git a/lib/connection-fsm.js b/lib/connection-fsm.js
index 5723d21..e78abf2 100644
--- a/lib/connection-fsm.js
+++ b/lib/connection-fsm.js
@@ -336,7 +336,7 @@ SocketMgrFSM.prototype.state_error = function (S) {
 };
 
 SocketMgrFSM.prototype.state_backoff = function (S) {
-	S.validTransitions(['failed', 'connecting']);
+	S.validTransitions(['failed', 'connecting', 'closed']);
 
 	/*
 	 * Unfortunately, "retries" actually means "attempts" in the cueball
